pandas/core/strings.py

```diff
@@ -109,9 +118,51 @@ def cat_safe(list_of_columns: List, sep: str):
 
 def _na_map(f, arr, na_result=np.nan, dtype=object):
     # should really _check_ for NA
+    if is_extension_array_dtype(arr.dtype):
+        arr = extract_array(arr)
+        return _map_ea(f, arr, na_value=na_result, dtype=dtype)
     return _map(f, arr, na_mask=True, na_value=na_result, dtype=dtype)
 
 
+def _map_ea(
+    func: Callable, arr: "StringArray", na_value: Any, dtype: Dtype
+) -> ArrayLike:
+    from pandas.arrays import IntegerArray, StringArray
+
+    mask = isna(arr)
+
+    assert isinstance(arr, StringArray)
+    arr = arr._ndarray
+
+    if is_integer_dtype(dtype):
+        na_value_is_na = isna(na_value)
+        if na_value_is_na:
+            na_value = 1
+        result = lib.map_infer_mask(
+            arr,
+            func,
+            mask.view("uint8"),
+            convert=False,
+            na_value=na_value,
+            dtype=np.dtype("int64"),
+        )
+
+        if not na_value_is_na:
+            mask[:] = False
+
+        return IntegerArray(result, mask)
+
+    elif is_string_dtype(dtype) and not is_object_dtype(dtype):
+        # i.e. StringDtype
+        result = lib.map_infer_mask(
+            arr, func, mask.view("uint8"), convert=False, na_value=na_value
+        )
+        return StringArray(result)
+    # TODO: BooleanArray
+    else:
```

Comment:
what actually hits this branch? IIUC, dtype is explicity passed as either Int64 or string, *never* object by the string routines, albeit you have a default of object.

Reply:
Right now, we hit this with the return types

1. bool (e.g. contains)
2. bytes (`encode`)
3. object -> list (findall, which returns an ndarray[list]
4. object -> Any (`get`)

We'll be able to avoid this for bool. Won't be able to avoid it for nested data or `.get`.

Reply:
can you indicate this with a comment.